File: 2775.py
class Pacote(object):
    def __init__(self, numero):
        self.numero = numero

    def set_tempo_troca(self, tempo_troca):
        self.tempo_troca = tempo_troca

    def __lt__(self, other):
        return self.numero > other.numero

    def mostrar(self, indice):
        print("PACOTE " + str(indice))
        print("  Numero: " + str(self.numero))
        print("  Tempo de troca: " + str(self.tempo_troca))


while True:
    try:
        qtd_pacotes = int(input())

        pacotes = [Pacote(int(n)) for n in str(input()).split()]
        tempos_troca = str(input()).split()

        for i in range(qtd_pacotes):
            pacotes[i].set_tempo_troca(int(tempos_troca[i]))

        soma = 0
        for i in range(0, qtd_pacotes):
            trocou = False
            for j in range(0, qtd_pacotes-1):
                if pacotes[j].__lt__(pacotes[j+1]):
                    pacotes[j], pacotes[j+1] = pacotes[j+1], pacotes[j]
                    soma = soma + pacotes[j].tempo_troca + pacotes[j+1].tempo_troca
                    trocou = True

            if not trocou:
                break

        print(soma)

    except EOFError:
        break

File: 2775.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Pacote {
public:
    int id;
    int tempo_troca;
};

bool comp(Pacote p1, Pacote p2) {
    return (p1.id < p2.id);
}

int main(int argc, char* argv[]) {
    int qtd_pacotes;

    while(cin >> qtd_pacotes) {
        vector<Pacote> pacotes(qtd_pacotes);

        for(int i=0;i<qtd_pacotes;++i) {
            cin >> pacotes[i].id;
        }

        for(int i=0;i<qtd_pacotes;++i) {
            cin >> pacotes[i].tempo_troca;
        }

        int tempo_total=0;
        for(int i=qtd_pacotes-1;i>0;--i) {
            for(int j=0;j<i;++j) {
                if(comp(pacotes[j+1], pacotes[j])) {
                    tempo_total += (pacotes[j].tempo_troca + pacotes[j+1].tempo_troca);
                    iter_swap(&pacotes[j], &pacotes[j+1]);
                }
            }
        }

        cout << tempo_total << endl;
    }

    return 0;
}
